src/handlers/transfers/handler.js

```diff
@@ -103,141 +105,92 @@ const prepare = async (error, messages) => {
     } else {
       message = messages
     }
-    let consumer
-    Logger.info('TransferService::prepare')
+    const payload = message.value.content.payload
+    const headers = message.value.content.headers
+    const action = message.value.metadata.event.action
+    const transferId = payload.transferId
     const kafkaTopic = message.topic
+    let consumer
+    Logger.info(Util.breadcrumb(location, { method: 'prepare' }))
     try {
       consumer = Kafka.Consumer.getConsumer(kafkaTopic)
-    } catch (e) {
+    } catch (err) {
       Logger.info(`No consumer found for topic ${kafkaTopic}`)
-      Logger.error(e)
+      Logger.error(err)
       histTimerEnd({ success: false, fspId: Config.INSTRUMENTATION_METRICS_LABELS.fspId })
       return true
     }
-    const payload = message.value.content.payload
-    const headers = message.value.content.headers
-
-    Logger.info('TransferService::prepare::checking for duplicates')
-    const { existsMatching, existsNotMatching } = await TransferService.validateDuplicateHash(payload)
+    // Action Letter (AL) is used to denote breadcrumb flow end
+    const AL = action === TEA.PREPARE ? 'P' : '?'
```

Comment:
For readability please rename to actionLetter also i would store 'P' and '?' under constants with a name that explains what it is

Reply:
ok